docs/en/concepts-and-designs/scope-definitions.md

```diff
@@ -29,6 +29,7 @@ Calculate the metrics data from each request of the service.
 | responseCode | Represent the response code of HTTP response, if this request is the HTTP call | | int|
 | type | Represent the type of each request. Such as: Database, HTTP, RPC, gRPC. | | enum |
 | tags | Represent the labels of each request and each value is made up with the `TagKey:TagValue` in the segment. | | `List<String>` |
+| sideCar.internalErrorCode | Represent the sidecar/proxy internal error code, the value bases on the implementation. | | string|
```

Comment:
FWIK, "proxy" is a more general name than `sidecar`. The latter one is often put together with "gateway" as Istio does, and therefore I prefer to use "proxy.internalErrorCode" instead of "sidecar.internalErrorCode".

Reply:
In the Istio concept, yes, I agree. But note here, we are adding a description for a service. If this is a gateway, as you mentioned, the service itself is the gateway, rather than this.
Also, for the gateway case, actually, there is no point to indicate whether sidecar issue or not, all errors are gateway's issue.

Reply:
As the attachment, I don't want it to be a `proxy`.

Reply:
> Also, for the gateway case, actually, there is no point to indicate whether sidecar issue or not, all errors are gateway's issue.

Definitely, a gateway is abstracted to service, but it still needs `sidecar.xxxx` to represent this dimension. It looks a little tricky, especially, the remarks emphasize `the sidecar/proxy internal error code`. if that, `the sidecar/gateway proxy internal error code` makes more sense to me.

Reply:
Fixed the comments and docs. Others are unchanged.